File: javascript/socketd/src/socketd/transport/core/ChannelSupporter.ts
import type {Processor} from "./Processor";
import type {Config} from "./Config";
import type {ChannelAssistant} from "./ChannelAssistant";

/**
 * 通道支持者（创建通道的参数）
 *
 * @author noear
 * @since 2.1
 */
export interface ChannelSupporter<S> {
    /**
     * 处理器
     */
    getProcessor(): Processor;

    /**
     * 配置
     */
    getConfig(): Config;

    /**
     * 通道助理
     */
    getAssistant(): ChannelAssistant<S>;
}

File: python/socketd/socketd/transport/core/ChannelSupporter.py
import typing
from abc import ABC, abstractmethod
from typing import TypeVar

from socketd.transport.core import Config
from socketd.transport.core.ChannelAssistant import ChannelAssistant
from socketd.transport.core.Processor import Processor

S = TypeVar("S")


class ChannelSupporter(typing.Generic[S], ABC):

    @abstractmethod
    def get_processor(self) -> Processor: ...

    @abstractmethod
    def get_config(self) -> Config: ...

    @abstractmethod
    def get_assistant(self) -> ChannelAssistant[S]: ...
